src/engine/Result.h

```diff
@@ -24,9 +23,16 @@
 // evaluated.
 class Result {
  private:
-  using Data = std::variant<IdTable, cppcoro::generator<IdTable>>;
-  // The actual entries. Needs to be mutable in order to consume a const entry.
-  mutable Data data_;
+  // Needs to be mutable in order to be consumable from a const result.
+  struct GenContainer {
+    mutable cppcoro::generator<IdTable> generator_;
+    mutable bool consumed_ = false;
```

Comment:
I think this bool must be atomic with acquire-release semantics, to make `const`` methods also thredsafe.